File: stress/t/arr_stencil.f
      PROGRAM ARRSTENCIL
      INTEGER*8 W,H,R,C,ACC,T,S,MX
      INTEGER*8 G(0:34),O(0:34)
      W=7
      H=5
      R=0
   10 IF (R.GE.H) GOTO 30
      C=0
   20 IF (C.GE.W) GOTO 25
      G(R*W+C)=MOD(R*W+C,9_8)
      C=C+1
      GOTO 20
   25 CONTINUE
      R=R+1
      GOTO 10
   30 CONTINUE
      R=0
   40 IF (R.GE.H) GOTO 70
      C=0
   50 IF (C.GE.W) GOTO 65
      ACC=0
      T=G(R*W+C)
      ACC=ACC+T
      IF (R.GT.0) THEN
        T=G((R-1)*W+C)
        ACC=ACC+T
      ENDIF
      IF (R.LT.H-1) THEN
        T=G((R+1)*W+C)
        ACC=ACC+T
      ENDIF
      IF (C.GT.0) THEN
        T=G(R*W+(C-1))
        ACC=ACC+T
      ENDIF
      IF (C.LT.W-1) THEN
        T=G(R*W+(C+1))
        ACC=ACC+T
      ENDIF
      O(R*W+C)=ACC
      C=C+1
      GOTO 50
   65 CONTINUE
      R=R+1
      GOTO 40
   70 CONTINUE
      S=0
      MX=0
      R=0
   80 IF (R.GE.H*W) GOTO 90
      T=O(R)
      S=S+T
      IF (T.GT.MX) MX=T
      R=R+1
      GOTO 80
   90 CONTINUE
      WRITE(*,'(A,I0)') 'stencil_sum=',S
      WRITE(*,'(A,I0)') 'stencil_max=',MX
      T=O(2*W+3)
      WRITE(*,'(A,I0)') 'center=',T
      T=O(0)
      WRITE(*,'(A,I0)') 'corner=',T
      END

File: stress/t/arr_stencil.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t w,h,r,c,acc,t,s,mx;
  int64_t G[35],O[35];
  w=7; h=5;
  for(r=0;r<h;r++){ for(c=0;c<w;c++){ G[r*w+c]=(r*w+c)%9; } }
  for(r=0;r<h;r++){
    for(c=0;c<w;c++){
      acc=0;
      t=G[r*w+c]; acc=acc+t;
      if(r>0){ t=G[(r-1)*w+c]; acc=acc+t; }
      if(r<h-1){ t=G[(r+1)*w+c]; acc=acc+t; }
      if(c>0){ t=G[r*w+(c-1)]; acc=acc+t; }
      if(c<w-1){ t=G[r*w+(c+1)]; acc=acc+t; }
      O[r*w+c]=acc;
    }
  }
  s=0; mx=0;
  for(r=0;r<h*w;r++){ t=O[r]; s=s+t; if(t>mx){ mx=t; } }
  printf("stencil_sum="); printf("%ld",(long)s); printf("\n");
  printf("stencil_max="); printf("%ld",(long)mx); printf("\n");
  t=O[2*w+3]; printf("center="); printf("%ld",(long)t); printf("\n");
  t=O[0]; printf("corner="); printf("%ld",(long)t); printf("\n");
  return 0;
}
